### PR TITLE
Adding parse logging for remote notifications.

### PR DESCRIPTION
**Pull request by MartinJibber**
I'm not 100% sure this is what we want, but here's a stab at it.

### CODE REVIEW

File: Shared/App/AppDelegate.swift [hunk 1]
```diff
@@ -41,6 +41,14 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
         print("DID FAIL TO REGISTER FOR PUSH \(error)")
     }
+
+    func application(_ application: UIApplication,
+                     didReceiveRemoteNotification userInfo: [AnyHashable : Any]) async -> UIBackgroundFetchResult {
+
+        EventLogger.trackRemoteNotification(userInfo: userInfo)
+
+        return .noData
+    }
```
Comment (MartinJibber) [inline Shared/App/AppDelegate.swift]: Is this the place to handle remote notification logging?

File: Shared/Managers/EventLogger.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        PFAnalytics.trackEvent(name)
+    }
+
+    static func trackRemoteNotification(userInfo: [AnyHashable : Any]) {
+        var dimensions: [String : String] = [:]
+
+        for kvp in userInfo {
+            guard let key = kvp.key as? String, let value = kvp.value as? String else { break }
+            dimensions[key] = value
+        }
```
Comment (MartinJibber) [inline Shared/Managers/EventLogger.swift]: Converting the user info into string data.

File: Shared/App/AppDelegate.swift (hunk 1 above)
Comment (Benjamindodgson) [reply Shared/App/AppDelegate.swift]: All notifications are handled in the notification extension

**Review by Benjamindodgson — CHANGES_REQUESTED** [review]
Add the logger to the extension

File: Notification Service/NotificationService.swift [hunk 3]
```diff
@@ -32,6 +32,8 @@ class NotificationService: UNNotificationServiceExtension {
         self.contentHandler = contentHandler
         self.request = request
 
+        EventLogger.trackRemoteNotification(userInfo: request.content.userInfo)
```
Comment (MartinJibber) [inline Notification Service/NotificationService.swift]: Logging events when the remote notification comes through.

**Review by Benjamindodgson — APPROVED** [review]

**Pull request merged**